lib/resty/shell.lua

```diff
@@ -11,14 +12,25 @@ local shell = {
     _VERSION = '0.01'
 }
 
-local sockproc_socket = "unix:/tmp/shell.sock"
-
+local default_socket = "unix:/tmp/shell.sock"
 
 function shell.execute(cmd, args)
     local timeout = args and args.timeout
     local input_data = args and args.data or ""
+    local socket = args and args.socket or default_socket
+
+    local is_tcp = true
+    if (find(socket, "unix:", 1, true)) then
```

Comment:
Oops, i missed this in the first reading: this will actually raise an error, if socket is a table.
I would do something like this instead:

local is_tcp  = (socket.host and tonumber(socket.port)) and true or false